### PR TITLE
test: adding an example for cluster on local pvc creation

### PR DESCRIPTION
**Pull request by rkachach**
**Description of your changes:**

adding an example for a cluster where local pvc are used
for mon store and for OSD creation

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                requests:
+                  storage: 20Gi
+              # IMPORTANT: Change the storage class depending on your environment
+              storageClassName: local-storage
+              volumeMode: Block
+              accessModes:
+                - ReadWriteOnce
+    # when onlyApplyOSDPlacement is false, will merge both placement.All() and storageClassDeviceSets.Placement
+    onlyApplyOSDPlacement: false
+  resources:
```
Comment (travisn) [inline deploy/examples/cluster-on-local-pvc-test.yaml:159]: Let's just remove the `resources` section from this example

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+                      - key: app
+                        operator: In
+                        values:
+                          - rook-ceph-osd
+                      - key: app
+                        operator: In
+                        values:
+                          - rook-ceph-osd-prepare
+                  topologyKey: kubernetes.io/hostname
+        resources:
```
Comment (travisn) [inline deploy/examples/cluster-on-local-pvc-test.yaml:134]: Remove `resources`

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    storage: 20Gi
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  # PV for mon must be a filesystem volume.
+  volumeMode: Filesystem
+  local:
+    # If you want to use dm devices like logical volume, please replace `/dev/sdb` with their device names like `/dev/vg-name/lv-name`.
+    path: /dev/vdb
+  nodeAffinity:
```
Comment (travisn) [inline deploy/examples/cluster-on-local-pvc-test.yaml:34]: Let's keep the example simple by removing all the `nodeAffinity`

**Review by travisn — CHANGES_REQUESTED** [review]
Just a few suggestions to keep this example simple for minikube.

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+volumeBindingMode: WaitForFirstConsumer
+---
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local0-0
+spec:
+  storageClassName: local-storage
+  capacity:
+    storage: 20Gi
```
Comment (sp98) [inline deploy/examples/cluster-on-local-pvc-test.yaml:25]: example creates both mon and OSD pvs with 20Gi. Suggesting to keep mon pv size to 10Gi.

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 5]
```diff
@@ -0,0 +1,174 @@
+#################################################################################################################
+# Define the settings for the rook-ceph cluster with common settings for a production cluster on top of bare metal.
+
+# This example expects three nodes, each with two available disks. Please modify it according to your environment.
+# See the documentation for more details on storage settings available.
+
+# For example, to create the cluster:
+#   kubectl create -f crds.yaml -f common.yaml -f operator.yaml
+#   kubectl create -f cluster-on-local-pvc.yaml
```
Comment (sp98) [inline deploy/examples/cluster-on-local-pvc-test.yaml:9]: `cluster-on-local-pvc.yaml` does not match the file name `cluster-on-local-pvc-test.yaml`

File: deploy/examples/cluster-on-local-pvc-test.yaml [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  storageClassName: local-storage
+  capacity:
+    storage: 20Gi
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  # PV for mon must be a filesystem volume.
+  volumeMode: Filesystem
+  local:
+    # If you want to use dm devices like logical volume, please replace `/dev/sdb` with their device names like `/dev/vg-name/lv-name`.
```
Comment (subhamkrai) [inline deploy/examples/cluster-on-local-pvc-test.yaml:32]:
```suggestion
    # To use dm devices like logical volume, please replace `/dev/sdb` with their device names like `/dev/vg-name/lv-name`.
```

same for below at L577

File: deploy/examples/cluster-on-local-pvc-test.yaml (hunk 3 above)
Comment (rkachach) [reply deploy/examples/cluster-on-local-pvc-test.yaml]: @travisn `nodeAffinity` seems to be a required parameter! .. when I remove it I get the following error:

`Error from server (Invalid): error when creating "mycluster.yaml": PersistentVolume "local0-0" is invalid: spec.nodeAffinity: Required value: Local volume requires node affinity`

**Review by travisn — CHANGES_REQUESTED** [review]
Looks good, please just squash the commits and add a commit description paragraph.

File: deploy/examples/cluster-on-local-pvc-test.yaml (hunk 4 above)
Comment (sp98) [reply deploy/examples/cluster-on-local-pvc-test.yaml]: Its still `20Gi` in the yaml file. Did we decide to keep it 20Gi?

File: deploy/examples/cluster-on-pvc-minikube.yaml [hunk 7]
```diff
@@ -0,0 +1,158 @@
...
+volumeBindingMode: WaitForFirstConsumer
+---
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+  name: local0-0
+spec:
+  storageClassName: local-storage
+  capacity:
+    storage: 20Gi
```
Comment (sp98) [inline deploy/examples/cluster-on-pvc-minikube.yaml:25]: This needs to be changed to 10Gi as well.

**Review by sp98 — CHANGES_REQUESTED** [review]

**Review by travisn — COMMENTED** [review]
@rkachach Could you also add a description paragraph to the commit? See the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure) for an example. It's helpful to have more details in the commit history.

**Review by travisn — APPROVED** [review]

**Review by subhamkrai — APPROVED** [review]

**Review by sp98 — APPROVED** [review]

**Pull request merged**